drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -766,7 +766,36 @@ static int set_up_plls(void)
 		stm32_clock_switch_to_hsi();
 		LL_RCC_SetAHBPrescaler(LL_RCC_SYSCLK_DIV_1);
 	}
+
+#if defined(CONFIG_STM32_MEMMAP) && defined(CONFIG_BOOTLOADER_MCUBOOT)
+	/*
+	 * Don't disable PLL during application initialization
+	 * that runs in memmap mode when (Q/O)SPI uses PLL
+	 * as its clock source.
+	 */
+#if defined(OCTOSPI1) || defined(OCTOSPI2)
+	if (LL_RCC_GetOSPIClockSource(LL_RCC_OSPI_CLKSOURCE) != LL_RCC_OSPI_CLKSOURCE_PLL1Q) {
+		LL_RCC_PLL1_Disable();
+	}
+	if (LL_RCC_GetOSPIClockSource(LL_RCC_OSPI_CLKSOURCE) != LL_RCC_OSPI_CLKSOURCE_PLL2R) {
+		LL_RCC_PLL2_Disable();
+	}
+#elif defined(QUADSPI)
+	if (LL_RCC_GetQSPIClockSource(LL_RCC_QSPI_CLKSOURCE) != LL_RCC_QSPI_CLKSOURCE_PLL1Q) {
+		LL_RCC_PLL1_Disable();
+	}
+	if (LL_RCC_GetQSPIClockSource(LL_RCC_QSPI_CLKSOURCE) != LL_RCC_QSPI_CLKSOURCE_PLL2R) {
+		LL_RCC_PLL2_Disable();
+	}
+#else
+	LL_RCC_PLL1_Disable();
+	LL_RCC_PLL2_Disable();
+#endif
+#else
 	LL_RCC_PLL1_Disable();
+	LL_RCC_PLL2_Disable();
+#endif
+	LL_RCC_PLL3_Disable();
```

Comment:
This whole piece of code ignores device tree and if (O|Q)SPI devices are enabled and if PLL are even required by applications, but that could be ok as I can understand considering all cases would make code much heavy and we miss required Zephyr apis to know clock consumers to really make it fully clean.

This being said I'd prefer this is put under each PLL code section bellow.

Reply:
It is placed here because:

* It is closer and relates to the note `Case of chan-loaded application:`
* We can't rely on application configuration as we don't know what the bootloader enabled.
  The bootloader might use PLL3, but the app won't. That makes `STM32_PLL3_ENABLE = 0`.
  
  Maybe we should rely on PLL registers (_`RCC_CR`:`PLL<n>ON` bits_) knowing we can't rely on DT when chain-loaded.
  What do you think?